### PR TITLE
Yamllint

### PR DESCRIPTION
**Pull request by synarete**
Require all yaml files to be well formatted, using 'yamllint' tool. If need, reformat yaml file using 'yq'.

### CODE REVIEW

File: hack/yq-fixup-yamls.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	dif=$(diff -q "${yaml}" "${yaml}".tmp)
+	if [ "${dif}" != "" ]; then
+		mv "${yaml}".tmp "${yaml}"
+	else
+		rm -f "${yaml}".tmp
+	fi
+}
+
+_yaml_fixup_files() {
+	yamls=$(find "$1" -type f -name '*.yml' -or -name '*.yaml')
```
Comment (phlogistonjohn) [inline hack/yq-fixup-yamls.sh:27]: Perhaps we should do -iname instead of name for case-insensitive matches? (example: FOO.YAML)

File: Makefile [hunk 2]
```diff
@@ -28,6 +28,7 @@ GO_CMD:=go
 GOFMT_CMD:=gofmt
 KUBECTL_CMD:=kubectl
 BUILDAH_CMD:=buildah
+YAMLLINT_CMD:=yamllint
```
Comment (phlogistonjohn) [inline Makefile:31]: Unlike YQ there's no rule to download yamllint if it is not found. Why's that?

**Review by phlogistonjohn — CHANGES_REQUESTED** [review]
Generally, looking good. I didn't dig deeply into the yamllint rules or the reformatted YAMLs and just assumed the tools were doing the right thing.

File: hack/yq-fixup-yamls.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+#   yq-fixup-yamls.sh dirpath
+#
+YQ=${YQ:-$(command -v yq)}
+${YQ} --version > /dev/null
+
+_yaml_fixup_file() {
+	yaml="$1"
+
+	# use yq as yaml formatter
+	${YQ} eval "${yaml}" > "${yaml}".tmp
```
[the remark this answers is not in the document]
Comment (synarete) [reply hack/yq-fixup-yamls.sh:15]: ACK

File: hack/yq-fixup-yamls.sh [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+	# override original yaml file if differ
+	dif=$(diff -q "${yaml}" "${yaml}".tmp)
```
[the remark this answers is not in the document]
Comment (synarete) [reply hack/yq-fixup-yamls.sh:18]: Indeed, I did not want to change timestamps if nothing changed. Re-thinking about it, I agree with you -- it confuses the naive reader. Will fix.

File: hack/yq-fixup-yamls.sh [hunk 5]
```diff
@@ -0,0 +1,40 @@
+#!/bin/sh
+# use yq to reformat yaml files to fit yamllint's syntax validity checks
+#
+# usage: 
+#   yq-fixup-yamls.sh file.yaml 
+#   yq-fixup-yamls.sh dirpath
+#
+YQ=${YQ:-$(command -v yq)}
+${YQ} --version > /dev/null
```
[the remark this answers is not in the document]
Comment (synarete) [reply hack/yq-fixup-yamls.sh:9]: Sanity check for the YQ variable. Apparently, it is confusing. I will remove it.

File: hack/yq-fixup-yamls.sh (hunk 1 above)
Comment (synarete) [reply hack/yq-fixup-yamls.sh:27]: Do we have such cases? IMHO, it would be better to rename FOO.YAML --> foo.yaml.

File: Makefile (hunk 2 above)
Comment (synarete) [reply Makefile:31]: [yamllint](https://github.com/adrienverge/yamllint) is a python utility. YAMLLINT_CMD follows the same rules as KUBECTL_CMD and BUILDAH_CMD: expects the user to install them in his PATH

File: hack/yq-fixup-yamls.sh [hunk 6]
```diff
@@ -0,0 +1,40 @@
+#!/bin/sh
```
[the remark this answers is not in the document]
Comment (synarete) [reply hack/yq-fixup-yamls.sh:1]: ACK

File: hack/yq-fixup-yamls.sh (hunk 1 above)
Comment (phlogistonjohn) [reply hack/yq-fixup-yamls.sh:27]: The issue is that it won't get caught. I'm not against preferring lowercase, I personally prefer it. So, while I won't insist on it I figured I would bring it up.

File: Makefile (hunk 2 above)
Comment (phlogistonjohn) [reply Makefile:31]: OK, thanks. Not for this PR, but we should probably think a little bit over how we handle our project dependencies/tools. I started an effort a couple of days ago but got bogged down "in the weeds". But I will probably revisit it soon.

File: hack/yq-fixup-yamls.sh [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+#   yq-fixup-yamls.sh dirpath
+#
+set -e
+YQ=${YQ:-$(command -v yq)}
+
+_yaml_fixup_file() {
+	yaml="$1"
+
+	${YQ} eval "${yaml}" > "${yaml}".tmp
+	mv "${yaml}".tmp "${yaml}"
```
Comment (raghavendra-talur) [inline hack/yq-fixup-yamls.sh:15]: There is an inplace option in yq if you prefer that approach.

Comment (synarete) [reply hack/yq-fixup-yamls.sh:15]: Missed this option... using '--inplace' feels better (avoid leaving '*.tmp' files behind). Will change.

File: hack/yq-fixup-yamls.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (phlogistonjohn) [reply hack/yq-fixup-yamls.sh]: The use of `set -e` earlier in the file should take care of this now.

**Review by phlogistonjohn — DISMISSED** [review]

**Review by phlogistonjohn — APPROVED** [review]

**Pull request merged**